mealie/services/group_services/shopping_lists.py

```diff
@@ -108,7 +118,23 @@ def remove_unused_recipe_references(self, shopping_list_id: UUID4) -> None:
         if list_refs_to_delete:
             self.list_refs.delete_many(list_refs_to_delete)
 
-    def bulk_create_items(self, create_items: list[ShoppingListItemCreate]) -> ShoppingListItemsCollectionOut:
+    def find_matching_label(self, item: ShoppingListItemBase) -> UUID4 | None:
+        if item.label_id:
+            return item.label_id
+        if item.food:
+            return item.food.label_id
```

Comment:
@michael-genson do I understand correctly that this will only be labelling "notes" and not "foods that don't already have labels"?

Reply:
I wonder if it might make sense to also do foods if their label is null? (helps those of us who haven't got all our data sorted yet!)

Reply:
The way this works is by fuzzy-matching whatever the user types in to the food store, and if it finds a match, assign it the label of that food. Because of this, we can't really do anything if it already has a food.

This PR basically helps label shopping list items that aren't parsed, or items that are entered manually (without a food). Since we still depend on the food store, we can't do anything for items that have a food already without some sort of external data.